File: turbine_array/closest_point.py
import numpy as np


def closest_point(points, lon, lat):

    point_list = np.array([lon,lat]).T

    closest_dist = ((point_list[:, 0] - points[:, 0, None])**2 +
                    (point_list[:, 1] - points[:, 1, None])**2)

    closest_point_indexes = np.argmin(closest_dist, axis=1)

    return closest_point_indexes

File: turbine_array/UTideCurrentVersion/closest_point.m
function [ closest_point_indexes] = closest_point(points, point_list )
%find the index of point in point list closest to the given points
%- assumes the points is size (m,2)  and point_list is size  (n,2)

for ii=1:length(points)
    [~,closest_point_indexes(ii)]=min((point_list(:,1)-points(ii,1)).^2+(point_list(:,2)-points(ii,2)).^2);
end

end

